librz/core/cbin.c

```diff
@@ -321,33 +345,90 @@ RZ_API int rz_core_bin_set_by_name(RzCore *core, const char *name) {
 	return false;
 }
 
-RZ_API int rz_core_bin_set_env(RzCore *r, RzBinFile *binfile) {
-	rz_return_val_if_fail(r, false);
-
-	RzBinObject *binobj = binfile ? binfile->o : NULL;
+RZ_API int rz_core_bin_apply_all_info(RzCore *r, RzBinFile *binfile) {
+	rz_return_val_if_fail(r && binfile, false);
+	RzBinObject *binobj = binfile->o;
 	RzBinInfo *info = binobj ? binobj->info : NULL;
-	if (info) {
-		int va = info->has_va;
-		const char *arch = info->arch;
-		ut16 bits = info->bits;
-		ut64 baseaddr = rz_bin_get_baddr(r->bin);
-		rz_config_set_i(r->config, "bin.baddr", baseaddr);
-		sdb_num_add(r->sdb, "orig_baddr", baseaddr, 0);
-		r->dbg->bp->baddr = baseaddr;
-		rz_config_set(r->config, "asm.arch", arch);
-		rz_config_set_i(r->config, "asm.bits", bits);
-		rz_config_set(r->config, "analysis.arch", arch);
-		if (info->cpu && *info->cpu) {
-			rz_config_set(r->config, "analysis.cpu", info->cpu);
+	if (!info) {
+		return false;
+	}
+	int va = info->has_va;
+	const char *arch = info->arch;
+	ut16 bits = info->bits;
+	ut64 baseaddr = rz_bin_get_baddr(r->bin);
+	rz_config_set_i(r->config, "bin.baddr", baseaddr);
+	sdb_num_add(r->sdb, "orig_baddr", baseaddr, 0);
+	r->dbg->bp->baddr = baseaddr;
+	rz_config_set(r->config, "asm.arch", arch);
+	rz_config_set_i(r->config, "asm.bits", bits);
+	rz_config_set(r->config, "analysis.arch", arch);
+	if (info->cpu && *info->cpu) {
+		rz_config_set(r->config, "analysis.cpu", info->cpu);
+	} else {
+		rz_config_set(r->config, "analysis.cpu", arch);
+	}
+	rz_asm_use(r->rasm, arch);
+
+	// ----
+	// inlined: rz_core_bin_info(r, RZ_CORE_BIN_ACC_ALL, NULL, RZ_MODE_SET, va, NULL, NULL);
+	ut64 loadaddr = binobj ? binobj->loadaddr : UT64_MAX;
+
+	// use our internal values for va
+	va = va ? VA_TRUE : VA_FALSE;
+
+	rz_core_bin_apply_strings(r, binfile);
```

Comment:
why all these calls?

Reply:
This is all the code that `rz_core_bin_info(r, RZ_CORE_BIN_ACC_ALL, NULL, RZ_MODE_SET, va, NULL, NULL)`, which was there before was calling. The plan is to make it into individual functions `rz_core_bin_apply_*()` per info type and only for applying info, while `rz_core_bin_info()` will only do printing.

Reply:
Ah, YES! If this is to split the set and the printing, then yes please :)

Reply:
Exactly this.